### PR TITLE
nasdf: Update (+ undocumented symbols check) to 1.3?

### PR DESCRIPTION
**Pull request by aartaka**
This adds:
- New compilation test: undocumented symbols (it caught the fact that `njson` conditions weren't documented!).
- Printing the compilation-tested packages to logs.

I'm also thinking of fixing SBCL-only type tests: `(subtype TYPE t)` seems to be a portable way to check whether the symbol is bound to a type. @Ambrevar, any caveats to this approach? Any reason not to use it?

### CODE REVIEW

**Comment by aartaka** [comment]
@Ambrevar, forget it, `subtype` doesn't work. However!
- CCL: `(typep t 'x) => SIMPLE-ERROR unknown type specifier: X`
- ECL: `(typep t 'x) => SIMPLE-ERROR X is not a valid type specifier.`
- SBCL: `(typep t 'x) => SIMPLE-ERROR unknown type specifier: X`.

So, even though spec says non-existent type specifier in `typep` is undefined behavior, three main implementations consistently error out, which allows for guessing whether the thing is a type specifier based on it not erroring in `typep`.

**Comment by aartaka** [comment]
More behaviors for `(typep t 'x)`:
- CLISP: error.
- GCL: returns `(SYSTEM::BAD-TYPE X) `.
- ABCL: returns NIL

Which means: we can make a type-testing functionlike:
``` lisp
(defun valid-type-p (type)
  (handler-case
      (progn
        (typep t type)
        t)
    (error () nil)))
```
and enable the unbound export tests for 
``` lisp
#+(or sbcl ccl ecl gcl)
```

File: nasdf/compilation-tests.lisp [hunk 1]
```diff
@@ -37,24 +42,48 @@ A sub-package has a name that starts with that of PACKAGE followed by a '/' sepa
...
+(defun list-undocumented-exports (package)
+  (let ((result '()))
+    (do-external-symbols (s (find-package package) result)
+      (unless (or (documentation s 'variable)
+                  (documentation s 'function)
+                  (documentation s 'compiler-macro)
+                  (documentation s 'setf)
+                  (documentation s 'method-combination)
+                  (documentation s 'type)
+                  (documentation s 'structure)
```
Comment (Ambrevar) [inline nasdf/compilation-tests.lisp:54]: Shorter:

```lisp
(handler-bind ((warning #'muffle-warning)) (some (lambda (doctype) (documentation 'foo doctype)) '(t  variable function ...)))

File: nasdf/compilation-tests.lisp [hunk 2]
```diff
@@ -37,24 +42,48 @@ A sub-package has a name that starts with that of PACKAGE followed by a '/' sepa
...
+                  (and (find-package :parenscript)
+                       (gethash s (symbol-value (find-symbol "*MACRO-TOPLEVEL*" :parenscript)))))
```
Comment (Ambrevar) [inline nasdf/compilation-tests.lisp:56]: Wait, you are not looking for the docstring here, are you?

**Review by Ambrevar — COMMENTED** [review]
Looking good overall, thanks!

File: nasdf/compilation-tests.lisp (hunk 2 above)
Comment (aartaka) [reply nasdf/compilation-tests.lisp:56]: No, I'm excluding the Parenscript symbols, because they are unlikely to have documentation.

**Comment by aartaka** [comment]
@Ambrevar, merging then?

Will propagate it to all of our libraries afterwards (`nfiles` has 33 undocumented symbols :P)

File: nasdf/compilation-tests.lisp (hunk 2 above)
Comment (Ambrevar) [reply nasdf/compilation-tests.lisp:51]: OK, maybe add a comment then, cause otherwise this is confusing.

**Comment by Ambrevar** [comment]
OK, and don't forget to propagate this to Nyxt and Ntemplate.
Thanks!

**Pull request merged**